plugins/flytekit-dask/flytekitplugins/dask/task.py

```diff
@@ -0,0 +1,109 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, Optional
+
+from flytekitplugins.dask import models
+from flytekitplugins.dask.resources import convert_resources_to_resource_model
+from google.protobuf.json_format import MessageToDict
+
+from flytekit import PythonFunctionTask, Resources
+from flytekit.configuration import SerializationSettings
+from flytekit.core.task import TaskPlugins
+
+
+@dataclass
+class JobPodSpec:
+    """
+    Configuration for the dask job runner pod
+
+    :param image: Custom image to use. If ``None``, will use the same image the task was registered with. Optional,
+        defaults to ``None``. The image must have ``dask[distributed]`` installed and should have the same Python
+        environment as the cluster (scheduler + worker pods).
+    :param requests: Resources to request for the job runner pod. If ``None``, the requests passed into the task will be
+        used. Optional, defaults to ``None``
+    :param limits: Resource limits for the job runner pod. If ``None``, the limits passed into the task will be used.
+        Optional, defaults to ``None``
+    """
+
+    image: Optional[str] = None
+    requests: Optional[Resources] = None
+    limits: Optional[Resources] = None
```

Comment:
Could we use the image, resource in the @task? In propeller, we can get the resource from taskSpec.
https://github.com/flyteorg/flytekit/blob/42f3de54a246ac2a00b1291b1ff8c881423d621f/flytekit/core/task.py#L85-L88

Reply:
Sounds good 👍 I will make those changes together with the ones discussed in https://github.com/flyteorg/flyteidl/pull/339